openerp/addons/base/ir/ir_model.py

```diff
@@ -732,7 +732,7 @@ def group_names_with_access(self, cr, model_name, access_mode):
         return [('%s/%s' % x) if x[0] else x[1] for x in cr.fetchall()]
 
     @tools.ormcache()
-    def check(self, cr, uid, model, mode='read', raise_exception=True, context=None):
+    def _check(self, cr, uid, model, mode='read', raise_exception=True):
```

Comment:
Actually, the `context` is useful for translating the error message but not for the cache key: in such a case, an exception is raised and the `ormcache` decorator does not catch the exception, and thus nothing is stored in the cache.